source/common/upstream/upstream_impl.cc

```diff
@@ -1698,7 +1701,16 @@ absl::Status ClusterImplBase::parseDropOverloadConfig(
         "Cluster drop_overloads config denominator setting is invalid : {}. Valid range 0~2.",
         drop_percentage.denominator()));
   }
-  drop_overload_ = UnitFloat(float(drop_percentage.numerator()) / (denominator));
+
+  // If DropOverloadRuntimeKey is not enabled, take the EDS drop_overload config.
+  // If DropOverloadRuntimeKey is enabled, take it.
+  float drop_ratio = float(drop_percentage.numerator()) / (denominator);
+  uint64_t drop_percent_runtime =
+      runtime_.snapshot().getInteger(ClusterImplBase::DropOverloadRuntimeKey, UINT32_MAX);
+  if (drop_percent_runtime != UINT32_MAX) {
+    drop_ratio = float(drop_percent_runtime) / float(100);
```

Comment:
This should be the minimum of the runtime value or config value.

Reply:
I think the runtime value, if enabled, means Envoy is in testing mode, so having it override the actual value make sense and easy to follow.  Choosing minimum of these two is kind of mixing testing mode and production mode, this will be hard to explain.  Please check the above comments as well: https://github.com/envoyproxy/envoy/pull/32062#discussion_r1467743312

Reply:
I agree that using the minimal value will be more difficult to comprehend and debug.

Reply:
After further thinking, it make sense to me to choose the min of the two.  Envoy should not choose a drop_overload value bigger than the EDS config to be safe.
BTW, this PR is closed. I created a new PR for this:
https://github.com/envoyproxy/envoy/pull/33737